events/README_S3_Object_Lambda.md

```diff
@@ -0,0 +1,78 @@
+# Sample Function
+
+The following is a sample class and Lambda function that receives Amazon S3 Object Lambda event record data as an input and returns an object metadata output.
+
+```go
+
+// main.go
+package main
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+func handler(ctx context.Context, event events.S3ObjectLambdaEvent) (*s3.WriteGetObjectResponseOutput, error) {
+	url := event.GetObjectContext.InputS3Url
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	transformedObject := TransformedObject{
+		Metadata: Metadata{
+			Length: len(bodyBytes),
+			Md5:    toMd5(bodyBytes),
+		},
+	}
+	jsonData, err := json.Marshal(transformedObject)
+	if err != nil {
+		return nil, err
+	}
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	svc := s3.NewFromConfig(cfg)
+	input := &s3.WriteGetObjectResponseInput{
+		RequestRoute: &event.GetObjectContext.OutputRoute,
+		RequestToken: &event.GetObjectContext.OutputToken,
+		Body:         strings.NewReader(string(jsonData)),
+	}
+	return svc.WriteGetObjectResponse(ctx, input)
```

Comment:
At a glance this doesn't look right as an example return value.

Looking at the docs https://docs.aws.amazon.com/AmazonS3/latest/userguide/olap-writing-lambda.html#olap-headobject

I see python functions returning things like `{statusCode: 200, <otherfields>}` and at least on java function defined with a `void` return type.

Reply:
Changed this function just to return error.